### PR TITLE
Fix/metrics

### LINKED ISSUE #40: metrics for ipam-controller
Hi! I've checked the source code and it seems that there are some metrics planned to be served; but if I'm not mistaken those are not yet served. Are there any plans to improve metrics setup? 

Also I was thinking that maybe it would be nice if we add another metric that shows the total ip range usage for each unique ip pool defined in the ClusterCIDR resources.

### PR DESCRIPTION
**Pull request by mneverov**
Requested metrics had existed, but were not exposed.

This patch:
- adds a new metrics server running on :9091 by default (see `IPAM_METRICS_ADDR` env. var)
- moves probe and metrics server to util
- replaces `k8s.io/component-base/metrics/legacyregistry` with `prometheus/client_golang`
- metrics are registered in an `init` func instead of using sync.Once

```sh
curl -v localhost:9091/metrics
# HELP node_ipam_controller_multicidrset_allocation_tries_per_request Histogram measuring CIDR allocation tries per request.
# TYPE node_ipam_controller_multicidrset_allocation_tries_per_request histogram
node_ipam_controller_multicidrset_allocation_tries_per_request_bucket{clusterCIDR="10.244.0.0/16",le="1"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_bucket{clusterCIDR="10.244.0.0/16",le="5"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_bucket{clusterCIDR="10.244.0.0/16",le="25"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_bucket{clusterCIDR="10.244.0.0/16",le="125"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_bucket{clusterCIDR="10.244.0.0/16",le="625"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_bucket{clusterCIDR="10.244.0.0/16",le="+Inf"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_sum{clusterCIDR="10.244.0.0/16"} 0
node_ipam_controller_multicidrset_allocation_tries_per_request_count{clusterCIDR="10.244.0.0/16"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_bucket{clusterCIDR="2001:db8::/110",le="1"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_bucket{clusterCIDR="2001:db8::/110",le="5"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_bucket{clusterCIDR="2001:db8::/110",le="25"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_bucket{clusterCIDR="2001:db8::/110",le="125"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_bucket{clusterCIDR="2001:db8::/110",le="625"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_bucket{clusterCIDR="2001:db8::/110",le="+Inf"} 3
node_ipam_controller_multicidrset_allocation_tries_per_request_sum{clusterCIDR="2001:db8::/110"} 0
node_ipam_controller_multicidrset_allocation_tries_per_request_count{clusterCIDR="2001:db8::/110"} 3
# HELP node_ipam_controller_multicidrset_cidrs_allocations_total Counter measuring total number of CIDR allocations.
# TYPE node_ipam_controller_multicidrset_cidrs_allocations_total counter
node_ipam_controller_multicidrset_cidrs_allocations_total{clusterCIDR="10.244.0.0/16"} 3
node_ipam_controller_multicidrset_cidrs_allocations_total{clusterCIDR="2001:db8::/110"} 3
# HELP node_ipam_controller_multicidrset_usage_cidrs Gauge measuring percentage of allocated CIDRs.
# TYPE node_ipam_controller_multicidrset_usage_cidrs gauge
node_ipam_controller_multicidrset_usage_cidrs{clusterCIDR="10.244.0.0/16"} 0.01171875
node_ipam_controller_multicidrset_usage_cidrs{clusterCIDR="2001:db8::/110"} 0.0029296875
# HELP node_ipam_controller_multicirdset_max_cidrs Maximum number of CIDRs that can be allocated.
# TYPE node_ipam_controller_multicirdset_max_cidrs gauge
node_ipam_controller_multicirdset_max_cidrs{clusterCIDR="10.244.0.0/16"} 256
node_ipam_controller_multicirdset_max_cidrs{clusterCIDR="2001:db8::/110"} 1024
```

Fixes #40

### CODE REVIEW

**Comment by mneverov** [comment]
@aojea ptal

**Comment by aojea** [comment]
> * adds a new metrics server running on :9091 by default (see `IPAM_METRICS_ADDR` env. var)

we found that opening multiple ports for expose different endpoints ends causing problems, specially in host network pods, what do you think if we add a new flag `--binding-address` or similar to indicate the endpoint where the healthz, livez and metrics are exposed and mark the existing one `health-probe-address` as deprecated.

This way we just maintain one server and use one port, using the url to discriminate

**Comment by mneverov** [comment]
> what do you think if ...

Good idea, wanted to combine those two initially, but then checked controller-runtime and it [has 3 webservers](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/manager/internal.go#L63): webhook, probe, and metrics.

File: main.go [hunk 1]
```diff
@@ -166,3 +166,11 @@ func runControllers(ctx context.Context, kubeClient kubernetes.Interface, cfg *r
 
 	nodeIpamController.Run(ctx)
 }
+
+func bindingAddress(cfg config) string {
+	if cfg.HealthProbeAddr != "" {
```
Comment (mneverov) [inline main.go:171]: By default `HealthProbeAddr` is empty now. Add this check for existing clusters.
The default addr value `:8081` from `HealthProbeAddr` is now specified for `WebserverBindAddr`

**Comment by aojea** [comment]
/lgtm

Thanks

**Pull request merged**